### PR TITLE
break ci out into jenkins stages

### PR DESCRIPTION
**Pull request by omgitsbillryan**
## Description of change
Breaks out all of the various stages we have in our `make ci` into their own jenkins stage.

### Why?
- Provides more insight into build times
- Stages may be "parallelized" in the future
- Provides faster feedback for failed builds in PRs
  - Rather than drilling into jenkins console output to see why the `Run Tests` stage failed, you will quickly see the name of the stage which failed

image

### Potential Drawbacks
Some overhead time incurred by reloading the ruby environment for each docker execution. I think that since CI is `RAILS_ENV=test`, this overhead is on the order of seconds.

#### Applies to all PRs

- [ ] Appropriate logging
- [ ] Swagger docs have been updated, if applicable
- [ ] Provide link to originating GitHub issue, or connected to it via ZenHub
- [ ] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [ ] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -19,14 +19,30 @@ ci:
...
 
-.PHONY: ci-spec
-ci-spec:
-	@$(BASH_TEST) "bin/rspec ${SPEC_PATH}"
-
 .PHONY: bash
 bash:
 	@$(COMPOSE_DEV) $(BASH)
 
+.PHONEY: ci-build
```
Comment (edmangimelli) [inline Makefile]:
```suggestion
.PHONY: ci-build
```

Comment (edmangimelli) [reply Makefile]: i see the same spelling (PHONEY) for `ci-down` too --or are they supposed to be like that?

Comment (omgitsbillryan) [reply Makefile]: whoops, nope. good eye 👍

**Comment by omgitsbillryan** [comment]
@nathanhruby I totally agree about the stable `ci` target. Perhaps I should simply remove it from make... In my changes to the `ci` **rake** task, I discovered that a number of developers are using this to verify their local changes. So I think that so long as the rake task is there, the make target can be deleted.

Keep an eye out for a separate PR for this.

File: Makefile (hunk 1 above)
Comment (nathanhruby) [reply Makefile]: That is most certainly my fault, thank you for catching it!

File: Makefile [hunk 2]
```diff
@@ -15,18 +15,34 @@ default: ci
 ci:
 	@$(BASH_TEST) "bin/rails db:setup db:migrate ci"
 
-.PHONEY: ci-down
+.PHONY: ci-down
```
Comment (kfrz) [inline Makefile]: :joy_cat:

**Pull request merged**